tests/WCPAY_UnitTestCase.php

```diff
@@ -61,8 +62,14 @@ function( $argument ) use ( $request_class, $request ) {
 						return get_class( $request ) === get_class( $argument ) || is_subclass_of( $argument, $request_class );
 					}
 				)
-			)
-			->willReturn( $response );
+			);
+
+		if ( ! is_null( $response ) ) {
+			$request
+				->expects( $this->exactly( $total_api_calls ) )
+				->method( 'format_response' )
+				->willReturn( new Response( $response ) );
```

Comment:
```suggestion
				->willReturn( $response instanceof Response ? $response : new Response( $response ) );
```

How about checking whether the object is already a response? This way we'd allow mocks of requests, which have custom responses.

Reply:
Nice addition, I've committed your suggestion!